src/Fixer/ClassNotation/VisibilityRequiredFixer.php

```diff
@@ -169,23 +175,30 @@ public function isCandidate(Tokens $tokens): bool
 
     protected function createConfigurationDefinition(): FixerConfigurationResolverInterface
     {
-        $elements = ['const', 'method', 'property'];
-
         return new FixerConfigurationResolver([
             (new FixerOptionBuilder('elements', 'The structural elements to fix (PHP >= 7.1 required for `const`).'))
                 ->setAllowedTypes(['string[]'])
-                ->setAllowedValues([new AllowedValueSubset($elements)])
-                ->setDefault($elements)
+                ->setAllowedValues([new AllowedValueSubset($this->elements)])
+                ->setDefault($this->elements)
                 ->getOption(),
         ]);
     }
 
+    protected function configurePostNormalisation(): void
+    {
+        $this->elements = $this->configuration['elements'];
+
+        if (\in_array('property', $this->elements, true)) {
+            $this->elements[] = 'promoted_property';
```

Comment:
if I cannot set property without enabling promoted_property, do i need both options?

either don't auto-enable promoted_property, or keep single option for both

Reply:
> do i need both options?

Who? Where?

> keep single option for both

Nothing has changed for the user, there is no change in docs.

The list `$this->elements` is for `TokensAnalyzer` only.

Reply:
I believe that:
```
$fixer->createConfigurationDefinition(); // definition built
$fixer->configure(..); $fixer->fix... ; // someone using the rule
$fixer->createConfigurationDefinition(); // different configuration definition! now, `promoted_property` is one of options !
```

Reply:
Oh my, I never thought anyone would call `createConfigurationDefinition` more than once. Fixed.